### ISSUE TITLE
App crashing due to TrueTime connection

### ISSUE THREAD

**Issue by NurLight19** [instacart/TrueTime.swift#19]
Hi,
I'm recently starting using your library. Now this kind of crash logs are coming from Crashlytics:

> Crashed: com.instacart.ntp.connection
> 0  TrueTime                       0x100a30e08 _TFFC8TrueTime13NTPConnectionP33_2A79C9048389EB5D5F57E571231D0EE311requestTimeFT_T_U_FT_T_ + 34
> 1  libswiftFoundation.dylib       0x10122ea08 _TFV10Foundation4Data15withUnsafeBytesu0_rfzFzGSPq__xx + 176
> 2  TrueTime                       0x100a34c5c _TTSf4g_g_n___TFFC8TrueTime13NTPConnectionP33_2A79C9048389EB5D5F57E571231D0EE314handleResponseFV10Foundation4DataT_U_FT_T_ + 228
> 3  TrueTime                       0x100a3473c _TPA__TFFC8TrueTime13NTPConnectionP33_2A79C9048389EB5D5F57E571231D0EE314handleResponseFV10Foundation4DataT_U_FT_T_ + 80
> 4  libdispatch.dylib              0x1878661fc _dispatch_call_block_and_release + 24
> 5  libdispatch.dylib              0x1878661bc _dispatch_client_callout + 16
> 6  libdispatch.dylib              0x1878743dc _dispatch_queue_serial_drain + 928
> 7  libdispatch.dylib              0x1878699a4 _dispatch_queue_invoke + 652
> 8  libdispatch.dylib              0x1878748d8 _dispatch_queue_override_invoke + 360
> 9  libdispatch.dylib              0x18787634c _dispatch_root_queue_drain + 572
> 10 libdispatch.dylib              0x1878760ac _dispatch_worker_thread3 + 124
> 11 libsystem_pthread.dylib        0x187a6f2a0 _pthread_wqthread + 1288
> 12 libsystem_pthread.dylib        0x187a6ed8c start_wqthread + 4

App tries to fetch NTP time from this URLs:
`let hostUrls: [URL] = [URL(string: "pool.ntp.org")!, URL(string: "time.apple.com")!]`

I couldn't find out the reason of the crash. Could you help me, please?

**Comment by solidsnack** [comment]
Maybe https://github.com/instacart/TrueTime.swift/blob/master/Sources/NTPConnection.swift#L240 needs a guard?

```swift
guard let packet = data.withUnsafeBytes { $0.pointee as? ntp_packet_t }?.nativeEndian
```

**Comment by NurLight19** [comment]
@msanders @solidsnack When will you guys fix this issue?

**Comment by msanders** [comment]
@NurLight19: Thank you for the report. Sorry for the delay, I was out of the office.

I believe this is due to receiving an empty packet from the server. Reproduced with this patch:

```
diff --git a/Sources/NTPConnection.swift b/Sources/NTPConnection.swift
index e7e5a59..83d7585 100644
--- a/Sources/NTPConnection.swift
+++ b/Sources/NTPConnection.swift
@@ -228,7 +228,7 @@ private extension NTPConnection {
         }
     }
 
-    func handleResponse(_ data: Data) {
+    func handleResponse(_ data2: Data) {
         let responseTicks = timeval.uptime()
         lockQueue.async {
             guard self.started else { return } // Socket closed.
@@ -237,6 +237,7 @@ private extension NTPConnection {
                 return
             }
 
+            let data = Data()
             let packet = data.withUnsafeBytes { $0.pointee as ntp_packet_t }.nativeEndian
             let responseTime = startTime.milliseconds + (responseTicks.milliseconds -
                                                          requestTicks.milliseconds)
```

Causes the following crash when running sample app (or tests) -

![screenshot 2017-02-13 13 01 02](https://cloud.githubusercontent.com/assets/54970/22905215/60a9824a-f1f4-11e6-981b-88ea40abf9d1.png)

Should be fixed by #21. Will be merging into a patch shortly.

**Comment by NurLight19** [comment]
@msanders Thank you too.
